ovs-service/src/main/java/org/dcsa/ovs/service/VesselScheduleService.java

```diff
@@ -1,21 +1,172 @@
 package org.dcsa.ovs.service;
 
+import lombok.Builder;
 import lombok.RequiredArgsConstructor;
-import org.dcsa.ovs.mapping.VesselScheduleMapper;
-import org.dcsa.ovs.persistence.repository.VesselScheduleRepository;
-import org.dcsa.ovs.mapping.ListMapper;
-import org.dcsa.ovs.transferobjects.VesselScheduleTO;
+import org.dcsa.ovs.persistence.entity.*;
+import org.dcsa.ovs.persistence.repository.ServiceRepository;
+import org.dcsa.ovs.transferobjects.*;
+import org.dcsa.ovs.transferobjects.enums.DimensionUnit;
+import org.dcsa.ovs.transferobjects.enums.PortCallStatusCode;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.Objects;
 
 @Service
 @RequiredArgsConstructor
 public class VesselScheduleService {
-  private final VesselScheduleRepository repository;
-  private final VesselScheduleMapper mapper;
 
-  public List<VesselScheduleTO> findAll() {
-    return ListMapper.convertList(repository.findAll(), mapper::toTO);
+  private final ServiceRepository serviceRepository;
+
+  @Builder
+  public static class ServiceSchedulesFilters {
+    String carrierServiceCode;
+    String universalServiceReference;
+    String vesselIMONumber;
+    String vesselName;
+    String voyageNumber;
+    String universalVoyageReference;
+    String unLocationCode;
+    String facilitySMDGCode;
+    String startDate;
+    String endDate;
+    Integer limit;
+    String cursor;
+    String apiVersion;
+  }
+
+  public List<VesselScheduleTO> findAll(ServiceSchedulesFilters requestFilters) {
+
+    return serviceRepository.findAll().stream()
+        .map(
+            service ->
+                VesselScheduleTO.builder()
+                    .carrierServiceName(service.getCarrierServiceName())
+                    .carrierServiceCode(service.getCarrierServiceCode())
+                    .universalServiceReference(service.getUniversalServiceReference())
+                    .vessels(extractVesselsToVesselTO(service.getVessels()))
+                    .build())
+        .toList();
+  }
+
+  private List<VesselTO> extractVesselsToVesselTO(List<Vessel> vessels) {
+    return vessels.stream()
+        .map(
+            vessel ->
+                VesselTO.builder()
+                    .vesselName(vessel.getName())
+                    .vesselIMONumber(vessel.getImoNumber())
+                    .vesselOperatorCarrierSMDGCode(vessel.getOperatorCarrier().getSmdgCode())
+                    .vesselCallSign(vessel.getCallSign())
+                    .vesselLength(vessel.getLength())
+                    .vesselWidth(vessel.getWidth())
+                    .dimensionUnit(
+                        DimensionUnit.valueFromString(vessel.getDimensionUnit()).orElse(null))
```

Comment:
If the value can't be mapped (and is not null) wouldn't something be horrible wrong and we should throw an exception?

Reply:
I would favor either an exception or a change to our API where we set a default dimensionUnit

Reply:
For now,  As discussed with @pimg it will be left as is.